From 35559da24eb2eef6d70ad6726b33af530252c769 Mon Sep 17 00:00:00 2001
From: Robin Gong <b38343@freescale.com>
Date: Tue, 25 Aug 2015 13:16:13 +0800
Subject: [PATCH 0462/5242] MLK-11407-1 soc: imx: gpc: enable PU bypass

commit  5f63df3390da1e32544e050e9f72aae2eb985504 from
https://source.codeaurora.org/external/imx/linux-imx.git

Enable PU bypass support on i.mx6 family.

Signed-off-by: Robin Gong <b38343@freescale.com>

Moved to new gpc implementation for 4.14.y

Signed-off-by: Leonard Crestez <leonard.crestez@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/soc/imx/gpc.c |   18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/drivers/soc/imx/gpc.c b/drivers/soc/imx/gpc.c
index 0097a93..818e373 100644
--- a/drivers/soc/imx/gpc.c
+++ b/drivers/soc/imx/gpc.c
@@ -400,6 +400,22 @@ static int imx_gpc_old_dt_init(struct device *dev, struct regmap *regmap,
 	return ret;
 }
 
+static void imx_gpc_handle_ldobypass(struct platform_device *pdev)
+{
+	struct regulator *pu_reg = imx_gpc_domains[GPC_PGC_DOMAIN_PU].supply;
+	u32 bypass = 0;
+	int ret;
+
+	ret = of_property_read_u32(pdev->dev.of_node, "fsl,ldo-bypass", &bypass);
+	if (ret && ret != -EINVAL)
+		dev_warn(&pdev->dev, "failed to read fsl,ldo-bypass property: %d\n", ret);
+
+	/* We only bypass pu since arm and soc has been set in u-boot */
+	if (pu_reg && bypass) {
+		regulator_allow_bypass(pu_reg, true);
+	}
+}
+
 static int imx_gpc_probe(struct platform_device *pdev)
 {
 	const struct of_device_id *of_id =
@@ -495,6 +511,8 @@ static int imx_gpc_probe(struct platform_device *pdev)
 		}
 	}
 
+	imx_gpc_handle_ldobypass(pdev);
+
 	return 0;
 }
 
-- 
1.7.9.5

